applications/CompressiblePotentialFlowApplication/tests/cpp_tests/test_processes.cpp

```diff
@@ -95,5 +96,66 @@ namespace Kratos {
       const int wake = pElement->GetValue(WAKE);
       KRATOS_CHECK_NEAR(wake, 1, 1e-6);
     }
+
+    KRATOS_TEST_CASE_IN_SUITE(ApplyFarFieldProcess, CompressiblePotentialApplicationFastSuite)
+    {
+      // Create model_part
+      Model this_model;
+      ModelPart& model_part = this_model.CreateModelPart("Main", 3);
+
+      // Variables addition
+      model_part.AddNodalSolutionStepVariable(VELOCITY_POTENTIAL);
+      model_part.AddNodalSolutionStepVariable(AUXILIARY_VELOCITY_POTENTIAL);
+
+      // Set model_part properties
+      BoundedVector<double, 3> free_stream_velocity = ZeroVector(3);
+      free_stream_velocity(0) = 10.0;
+      model_part.GetProcessInfo()[FREE_STREAM_VELOCITY] = free_stream_velocity;
+
+      // Create nodes
+      model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+      model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+      model_part.CreateNewNode(3, 0.0, 1.0, 0.0);
+      model_part.CreateNewNode(4, 1.0, 1.0, 0.0);
+
+      for (auto& r_node : model_part.Nodes()) {
+        r_node.AddDof(VELOCITY_POTENTIAL);
+      }
+
+      model_part.CreateNewProperties(0);
+      Properties::Pointer pProp = model_part.pGetProperties(0);
+
+      std::vector<ModelPart::IndexType> cond1{1, 2};
+      std::vector<ModelPart::IndexType> cond2{2, 4};
+      std::vector<ModelPart::IndexType> cond3{4, 3};
+      std::vector<ModelPart::IndexType> cond4{3, 1};
+
+      model_part.CreateNewCondition("Condition2D2N", 1, cond1, pProp);
+      model_part.CreateNewCondition("Condition2D2N", 2, cond2, pProp);
+      model_part.CreateNewCondition("Condition2D2N", 3, cond3, pProp);
+      model_part.CreateNewCondition("Condition2D2N", 4, cond4, pProp);
+
+
+      // Set initial potential
+      const double initial_potential = 1.0;
+
+      // Construct the ApplyFarFieldProcess
+      ApplyFarFieldProcess ApplyFarFieldProcess(model_part, initial_potential);
+
+      // Execute the ApplyFarFieldProcess
+      ApplyFarFieldProcess.Execute();
+      ApplyFarFieldProcess.InitializeFlowField();
```

Comment:
Do you always need to call the `InitializeFlowField` after the `Execute`? If this is the case, I think this call should be included inside the `Execute` method. If not, shouldn't it be a different process? (note that by doing this you are exposing a method which doesn't exist in the base Process interface).

Reply:
No not always, its optional and the flag is checked here:

https://github.com/KratosMultiphysics/Kratos/blob/809ab0554a5bf84afde6dbd9a1d3ca3bef0694b0/applications/CompressiblePotentialFlowApplication/python_scripts/apply_far_field_process.py#L59-L64

If exposing methods that are not from the base Process interface is not desired, it can definitely be a different process.

I can also include the InitializeFlowField in the Execute method of the CPP and include a boolean in the constructor, so the python interface would look like this:

```
def ExecuteInitializeSolutionStep(self): 
     far_field_process=CPFApp.ApplyFarFieldProcess(self.far_field_model_part, self.inlet_potential_0, self.initialize_flow_field) 
     far_field_process.Execute() 

```
Also I can include the check that itsin the CPP method and a boolean to

Reply:
I went for the second option in the end, so there are no methods exposed that not exist in the base Process interface